### PR TITLE
Fix for switch debug & release builds invalidates build

### PR DESCRIPTION
**Pull request by prachipai**

### CODE REVIEW

File: Sources/Build/llbuild.swift [hunk 1]
```diff
@@ -304,20 +303,28 @@ public struct LLBuildManifestGenerator {
...
-    public var llbuildTargetName: String {
-        return "\(name).module"
+    public func getCommandName(config: String?) -> String {
+       return "C." + getBuildTargetName(config: config)
     }
-
-    var commandName: String {
-        return "C.\(llbuildTargetName)"
+    
+    public func getBuildTargetName(config: String?) -> String {
```
Comment (aciidb0mb3r) [inline Sources/Build/llbuild.swift]: This shouldn't be optional. We should always pass the config.

Comment (aciidb0mb3r) [reply Sources/Build/llbuild.swift]: nit: function name -> `getLLBuildTargetName(config:)`

File: Sources/Build/llbuild.swift [hunk 2]
```diff
@@ -304,20 +303,28 @@ public struct LLBuildManifestGenerator {
...
+        if let config = config {
+            if config == "release" {
+                return "\(name)-release.module"
+            }
+            else if config == "debug" {
+                return "\(name)-debug.module"
```
Comment (aciidb0mb3r) [inline Sources/Build/llbuild.swift]: Once the config is non-optional, this function can just be: `return "\(name)-\(config).module"`

File: Sources/Build/llbuild.swift (hunk 1 above)
Comment (prachipai) [reply Sources/Build/llbuild.swift]: Done

File: Sources/Build/llbuild.swift (hunk 2 above)
Comment (prachipai) [reply Sources/Build/llbuild.swift]: Done

File: Sources/Build/llbuild.swift [hunk 3]
```diff
@@ -308,20 +307,23 @@ public struct LLBuildManifestGenerator {
...
     }
-
-    var commandName: String {
-        return "C.\(llbuildTargetName)"
+    
+    public func getLLBuildTargetName(config: String) -> String {
+        if config == "release" {
+            return "\(name)-release.module"
+        }
+        return "\(name)-debug.module"
```
Comment (aciidb0mb3r) [inline Sources/Build/llbuild.swift]: This can just be `return "\(name)-\(config).module"

Comment (prachipai) [reply Sources/Build/llbuild.swift]: done

File: Sources/Build/llbuild.swift [hunk 4]
```diff
@@ -267,11 +266,11 @@ public struct LLBuildManifestGenerator {
             }
         }
 
-        var buildTarget = Target(name: target.target.llbuildTargetName)
+        var buildTarget = Target(name: target.target.getLLBuildTargetName(config: plan.buildParameters.configuration.dirname))
```
[the remark this answers is not in the document]
Comment (prachipai) [reply Sources/Build/llbuild.swift]: done

File: Sources/Build/llbuild.swift [hunk 5]
```diff
@@ -266,12 +266,12 @@ public struct LLBuildManifestGenerator {
                 }
             }
         }
-
```
Comment (aciidb0mb3r) [inline Sources/Build/llbuild.swift]: nit: should keep this newline

File: Sources/Commands/SwiftTool.swift [hunk 6]
```diff
@@ -564,29 +564,29 @@ public class SwiftTool<Options: ToolOptions> {
     func getManifestLoader() throws -> ManifestLoader {
         return try _manifestLoader.dematerialize()
     }
-
-    func computeLLBuildTargetName(for subset: BuildSubset) throws -> String? {
+
```
Comment (aciidb0mb3r) [inline Sources/Commands/SwiftTool.swift]: nit: same space issue here

**Comment by prachipai** [comment]
Added for products too!

**Comment by ddunbar** [comment]
Is this ready for rereview?

**Comment by prachipai** [comment]
We're thinking to write some tests to make sure other things aren't getting broken.  Ankit was worried about this.

**Comment by aciidb0mb3r** [comment]
This looks good.

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by ddunbar** [comment]
Yay yay yay! Can’t wait for a toolchain...

**Pull request merged**